src/solvers/dgsem_tree/dg_3d.jl

```diff
@@ -196,6 +240,68 @@ function calc_volume_integral!(du, u,
 end
 
 
+# Calculate 3D twopoint flux (element version)
+@inline function calcflux_twopoint!(f1, f2, f3, u, element,
+                                    mesh::TreeMesh{3}, equations, volume_flux, dg::DG, cache)
+
+  for k in eachnode(dg), j in eachnode(dg), i in eachnode(dg)
+    # Pull the solution values at the node i,j,k
+    u_node = get_node_vars(u, equations, dg, i, j, k, element)
+    # diagonal (consistent) part not needed since diagonal of
+    # dg.basis.derivative_split_transpose is zero!
+    set_node_vars!(f1, zero(u_node), equations, dg, i, i, j, k)
+    set_node_vars!(f2, zero(u_node), equations, dg, j, i, j, k)
+    set_node_vars!(f3, zero(u_node), equations, dg, k, i, j, k)
+
+    # Flux in x-direction
+    for ii in (i+1):nnodes(dg)
+      u_ll = get_node_vars(u, equations, dg, i,  j, k, element)
+      u_rr = get_node_vars(u, equations, dg, ii, j, k, element)
+      flux = volume_flux(u_ll, u_rr, 1, equations) # 1-> x-direction
+      set_node_vars!(f1, flux, equations, dg, i, ii, j, k)
+      set_node_vars!(f1, flux, equations, dg, ii, i, j, k)
+    end
+
+    # Flux in y-direction
+    for jj in (j+1):nnodes(dg)
+      u_ll = get_node_vars(u, equations, dg, i, j,  k, element)
+      u_rr = get_node_vars(u, equations, dg, i, jj, k, element)
+      flux = volume_flux(u_ll, u_rr, 2, equations) # 2 -> y-direction
+      set_node_vars!(f2, flux, equations, dg, j, i, jj, k)
+      set_node_vars!(f2, flux, equations, dg, jj, i, j, k)
+    end
+
+    # Flux in z-direction
+    for kk in (k+1):nnodes(dg)
+      u_ll = get_node_vars(u, equations, dg, i, j, k,  element)
+      u_rr = get_node_vars(u, equations, dg, i, j, kk, element)
+      flux = volume_flux(u_ll, u_rr, 3, equations) # 3 -> z-direction
+      set_node_vars!(f3, flux, equations, dg, k, i, j, kk)
+      set_node_vars!(f3, flux, equations, dg, kk, i, j, k)
+    end
+  end
+
+  calcflux_twopoint_nonconservative!(f1, f2, f3, u, element,
+                                     have_nonconservative_terms(equations),
+                                     mesh, equations, dg, cache)
+end
+
+function calcflux_twopoint_nonconservative!(f1, f2, f3, u, element,
+                                            nonconservative_terms::Val{false},
+                                            mesh::Union{TreeMesh{3}, StructuredMesh{3}},
+                                            equations, dg::DG, cache)
```

Comment:
Merge conflict again, every instance of `calcflux_twopoint_nonconservative!` should be gone now